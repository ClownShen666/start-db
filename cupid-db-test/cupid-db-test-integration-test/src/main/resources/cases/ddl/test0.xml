<?xml version="1.0" encoding="utf-8" ?>
<testcase>

    <xml>test0.xml</xml>

    <case><!--  求普通对数，num > 0  log(base: Double, num: Double): Double  -->
        <sql type="query">select log(?, ?);</sql>
        <assertion>[2][2]</assertion>
        <assertion>[2][4]</assertion>
        <assertion>[0][2]</assertion>  底数可以为0
        <assertion error="输入不能为0">[2][0]</assertion>
        <assertion error="输入不能为负">[2][-2]</assertion>
    </case>

    <case><!--  绝对值  abs(num: Double): Double  -->
        <sql type="query">select abs(?);</sql>  返回值类型不为double
        <assertion>[-100]</assertion>
        <assertion>[0]</assertion>
        <assertion>[100.123]</assertion>
        <assertion error="输入不能为空">[null]</assertion>
    </case>
</testcase>