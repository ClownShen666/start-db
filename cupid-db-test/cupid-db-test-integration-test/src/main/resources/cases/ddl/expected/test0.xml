<?xml version="1.0" encoding="UTF-8" ?>

<dataset>

    <sql><!-- select log(2, 2); -->
        <column name="EXPR$0" type="double"/>
        <row>1.0</row>
    </sql>

    <sql><!-- select log(2, 4); -->
        <column name="EXPR$0" type="double"/>
        <row>2.0</row>
    </sql>

    <sql><!-- select log(0, 2); -->
        <column name="EXPR$0" type="double"/>
        <row>0.0</row>
    </sql>

    <sql><!-- select abs(-100)-->
        <column name="EXPR$0" type="int"/>
        <row>100</row>
    </sql>

    <sql><!-- select abs(0)-->
        <column name="EXPR$0" type="int"/>
        <row>0</row>
    </sql>

    <sql><!-- select abs(100.123)-->
        <column name="EXPR$0" type="decimal"/>
        <row>100.123</row>
    </sql>

</dataset>
